src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -208,6 +215,40 @@ public boolean accept(Path path) {
     }
 
     /**
+     * Ensure reads in a pair fall in the same partition (input split), if the reads are queryname-sorted,
+     * so they are processed together. No shuffle is needed.
+     */
+    JavaRDD<GATKRead> putPairsInSamePartition(final SAMFileHeader header, final JavaRDD<GATKRead> reads) {
```

Comment:
Sorry, I didn't explained it well. I would like to have in the same partition unmapped reads with the same read name if they are sorted by queryname. What I meant is that could be a public static method here or somewhere as an API to use outside `ReadSparkSource`...